File: Practice/atcoder/ABC/059/src/c.go
package main

import (
	"fmt"
)

func sum(a []int) (ret int) {
	for _, v := range a {
		ret += v
	}
	return
}

func main() {
	var n int
	fmt.Scan(&n)
	arr := make([]int, n)
	for i := 0; i < n; i++ {
		var x int
		fmt.Scan(&x)
		arr[i] = x
	}
	// なんとなくiまでとi-1までの和が1と-1が交互になるようにしてnが奇数だったら最初の数値には手を付けないみたいな感じ
	// i項までの和が0になったらいまみてる数値の符号だけインクリメントかデクリメントする
	res := append([]int{}, arr...)
	first := arr[0]
	ans := 0
	var minus bool
	if first < 0 {
		minus = true
	} else {
		minus = false
	}
	for i, v := range arr[1:] {
		/*
			i=1 -1
			i=2	-1+4=3
			i=3 3+3(-7)=-1 7回インクリメント
			i=4 -1+2 = 1
			i=5 1-5 = -4
			i=6 -4+4(+1) = 1 1回インクリメント
			結果
			-1 4 -7 1 -5 1
		*/
		var sum int
		if minus{
			if 0<sum(arr[:i+1]){
				minus != minus
				continue
			}else{
				
			}
		}
	}

}

File: Practice/atcoder/ABC/059/src/c.py
# 気づき
# ①+-+-+-..か-+-+-+...のふたつを試して小さい方
# ②満たさなくなったら帳尻を合わせる（いまみてるやつだけでいい)
# ③1<=i<=nに対して和が0でない -> a1≠0（これは①とかぶる）
# 累積和をとってインデックスの偶奇で判断

n = int(input())
arr = list(map(int, input().split()))
p, m = 0, 0  # 累積和
pn, mn = 0, 0  # （操作した回数の）カウント
for i in range(n):
    if i % 2 == 0:
        # pは正にしたい
        if p + arr[i] > 0:
            p += arr[i]
        else:
            # 少ない操作で和を大きくしたいので1との差を取る
            d = p + arr[i]  # 差
            pn += abs(1 - d)
            p = 1
    else:
        # pは負にしたい
        if p + arr[i] < 0:
            p += arr[i]
        else:
            # 少ない操作で和を小さくしたいので-1との差をとる
            d = p + arr[i]
            pn += abs(-1 - d)
            p = -1
for i in range(n):
    if i % 2 == 0:
        # mは負にしたい
        if m + arr[i] < 0:
            m += arr[i]
        else:
            # ry
            d = m + arr[i]
            mn += abs(-1 - d)
            m = -1
    else:
        # mは正にしたい
        if m + arr[i] > 0:
            m += arr[i]
        else:
            # ry
            d = m + arr[i]
            mn += abs(1 - d)
            m = 1

print(min(pn, mn))
